pkgs/development/libraries/protobuf/generic-v3.nix

```diff
@@ -1,60 +1,68 @@
-{ stdenv
-, fetchFromGitHub
-, autoreconfHook, zlib, gmock, buildPackages
+{ stdenv, fetchFromGitHub, lib
+, zlib, cmake, gmock
 , version, sha256
 , ...
 }:
-
-let
-mkProtobufDerivation = buildProtobuf: stdenv: stdenv.mkDerivation {
+stdenv.mkDerivation rec {
+  # make sure you test also -A pythonPackages.protobuf
   pname = "protobuf";
   inherit version;
-
-  # make sure you test also -A pythonPackages.protobuf
   src = fetchFromGitHub {
     owner = "protocolbuffers";
     repo = "protobuf";
     rev = "v${version}";
     inherit sha256;
   };
 
-  postPatch = ''
-    rm -rf gmock
-    cp -r ${gmock.src}/googlemock gmock
-    cp -r ${gmock.src}/googletest googletest
-    chmod -R a+w gmock
-    chmod -R a+w googletest
-    ln -s ../googletest gmock/gtest
-  '' + stdenv.lib.optionalString stdenv.isDarwin ''
-    substituteInPlace src/google/protobuf/testing/googletest.cc \
-      --replace 'tmpnam(b)' '"'$TMPDIR'/foo"'
-  '';
+  patches =
+    [ ./protobuf-cmake.patch ]
+    # In version 3.6.X,
+    # CMake doesn't generate the version.rc file from version.rc.in.
+    ++ lib.optional
+       (lib.versionAtLeast version "3.6" && lib.versionOlder version "3.7")
+       ./protobuf-3.6-cmake-versionrc.patch;
```

Comment:
Upstream only generates that one for windows:
https://github.com/protocolbuffers/protobuf/blob/7b8a241b575b756c4dd7a5aec7f7a60854d5f711/cmake/CMakeLists.txt#L219-L220
Why do we need it? If we need it, we should let upstream know by submitting the patch or filing an issue.

Reply:
It causes the build to fail, but just for version 3.6, if we don't generate it, thus the version bounding check.